instrumentation/lettuce/lettuce-5.1/src/test/groovy/io/opentelemetry/auto/instrumentation/lettuce/v5_1/LettuceAsyncClientTest.groovy

```diff
@@ -0,0 +1,406 @@
+import io.lettuce.core.api.async.RedisAsyncCommands
+import io.lettuce.core.api.sync.RedisCommands
+import io.lettuce.core.codec.StringCodec
+import io.opentelemetry.auto.test.AgentTestRunner
+import io.opentelemetry.auto.test.utils.PortUtils
+import redis.embedded.RedisServer
+import spock.lang.Shared
+import spock.util.concurrent.AsyncConditions
+
+import java.util.concurrent.ExecutionException
+import java.util.concurrent.TimeUnit
+import java.util.function.BiConsumer
+import java.util.function.BiFunction
+import java.util.function.Consumer
+import java.util.function.Function
+
+import static io.opentelemetry.trace.Span.Kind.CLIENT
+
+class LettuceAsyncClientTest extends AgentTestRunner {
+  public static final String HOST = "127.0.0.1"
+  public static final int DB_INDEX = 0
+  // Disable autoreconnect so we do not get stray traces popping up on server shutdown
+  public static final ClientOptions CLIENT_OPTIONS = ClientOptions.builder().autoReconnect(false).build()
+
+  @Shared
+  int port
+  @Shared
+  int incorrectPort
+  @Shared
+  String dbAddr
+  @Shared
+  String dbAddrNonExistent
+  @Shared
+  String dbUriNonExistent
+  @Shared
+  String embeddedDbUri
+
+  @Shared
+  RedisServer redisServer
+
+  @Shared
+  Map<String, String> testHashMap = [
+    firstname: "John",
+    lastname : "Doe",
+    age      : "53"
+  ]
+
+  RedisClient redisClient
+  StatefulConnection connection
+  RedisAsyncCommands<String, ?> asyncCommands
+  RedisCommands<String, ?> syncCommands
+
+  def setupSpec() {
+    port = PortUtils.randomOpenPort()
+    incorrectPort = PortUtils.randomOpenPort()
+    dbAddr = HOST + ":" + port + "/" + DB_INDEX
+    dbAddrNonExistent = HOST + ":" + incorrectPort + "/" + DB_INDEX
+    dbUriNonExistent = "redis://" + dbAddrNonExistent
+    embeddedDbUri = "redis://" + dbAddr
+
+    redisServer = RedisServer.builder()
+    // bind to localhost to avoid firewall popup
+      .setting("bind " + HOST)
+    // set max memory to avoid problems in CI
+      .setting("maxmemory 128M")
+      .port(port).build()
+  }
+
+  def setup() {
+    redisClient = RedisClient.create(embeddedDbUri)
+
+    println "Using redis: $redisServer.args"
+    redisServer.start()
+    redisClient.setOptions(CLIENT_OPTIONS)
+
+    connection = redisClient.connect()
+    asyncCommands = connection.async()
+    syncCommands = connection.sync()
+
+    syncCommands.set("TESTKEY", "TESTVAL")
+
+    // 1 set
+    TEST_WRITER.waitForTraces(1)
+    TEST_WRITER.clear()
+  }
+
+  def cleanup() {
+    connection.close()
+    redisServer.stop()
+  }
+
+  def "connect using get on ConnectionFuture"() {
+    setup:
+    RedisClient testConnectionClient = RedisClient.create(embeddedDbUri)
+    testConnectionClient.setOptions(CLIENT_OPTIONS)
+
+    when:
+    ConnectionFuture connectionFuture = testConnectionClient.connectAsync(StringCodec.UTF8,
+      new RedisURI(HOST, port, 3, TimeUnit.SECONDS))
+    StatefulConnection connection = connectionFuture.get()
```

Comment:
I don't think that all our tests has inherent timeout. At least I have seen test hanging for a very long time before I abort them manually.

Reply:
Hmm - unfortunately `ConnectionFuture` doesn't offer any method that accepts a timeout :( Can I leave it like this since it's the same issue as the `lettuce-5.0` instrumentation?